types/src/model/mod.rs

```diff
@@ -65,3 +69,33 @@ pub use self::{
         UnloadWallet, WalletCreateFundedPsbt, WalletDisplayAddress, WalletProcessPsbt,
     },
 };
+
+/// Models the data returned by Core for a script pubkey.
+///
+/// This is used by methods in the blockchain section and in the raw transaction section (i.e raw
+/// transaction and psbt methods). The shape changed in Core v22 but the new shape is fully
+/// backwards compatible so we only provide it not a v0.17 specific type.
+#[derive(Clone, Debug, PartialEq, Eq, Deserialize, Serialize)]
+#[cfg_attr(feature = "serde-deny-unknown-fields", serde(deny_unknown_fields))]
+pub struct ScriptPubkey {
+    /// Script assembly.
+    pub asm: String,
+    /// Inferred descriptor for the output. v23 and later only.
+    pub descriptor: Option<String>,
+    /// Script hex.
+    pub hex: ScriptBuf,
```

Comment:
I'd put this as the first field and call it `script`, or `script_pubkey`, or `inner`. Anything but `hex`, which it no longer is.